schema/crdb/2.0.0/up.sql

```diff
@@ -0,0 +1,25 @@
+-- CRDB documentation recommends the following:
+--  "Execute schema changes either as single statements (as an implicit transaction),
+--   or in an explicit transaction consisting of the single schema change statement."
+--
+-- For each schema change, we transactionally:
+-- 1. Check the current version
+-- 2. Apply the idempotent update
+
+BEGIN;
+
+SELECT CAST(
+    IF(
+        (
+            SELECT version = '1.0.0' and target_version = '2.0.0'
+            FROM omicron.public.db_metadata WHERE singleton = true
+        ),
+        'true',
+        'Invalid starting version for schema change'
+    ) AS BOOL
+);
+
```

Comment:
Should this be part of the framework instead of having people copy/paste it into each up.sql?

Reply:
I went back and forth on this. On the one hand: Yes, it would be really nice to cut out this boilerplate. On the other hand, the conditions of "what can someone call from within a transaction" are pretty strict -- I don't want anyone to accidentally write an `up.sql` file that includes multiple DDL statements. I think it would be more correct, in these situations, to have one transaction per DDL statement.

TL;DR: I would like this, but I haven't figured out a good way to safely hide the details of "operating within a single transaction".